reader_index.go

```diff
@@ -46,6 +46,7 @@ func (f *rsfReader) ReadIndex(r io.Reader) (Index, error) {
 	// index version.
 	if bytes.Equal(header, IndexVersion2) {
 		f.indexVersion = 2
+		f.pos += 3
```

Comment:
The main source of the bug was that we were not incrementing the reader position here, so it remained 3 bytes behind for `V2` manifests. For V1 manifests, we increment by `4`, below, so it was OK.